### PR TITLE
Use concat-to-dir instead of concat

### PR DESCRIPTION
**Pull request by adithyaov**
@bastibe Thank you for this wonderful package. It's very clean and elegant. 

This PR contains a very small change interpreting a few variables as directories.

### CODE REVIEW

File: org-static-blog.el [hunk 1]
```diff
@@ -438,7 +442,7 @@ Preamble and Postamble are excluded, too."
 For example, when `org-static-blog-publish-url` is set to 'https://example.com/'
 and `relative-url` is passed as 'archive.html' then the function
 will return 'https://example.com/archive.html'."
-  (concat org-static-blog-publish-url relative-url))
+  (concat-to-dir org-static-blog-publish-url relative-url))
```
Comment (bastibe) [inline org-static-blog.el:445]: `org-static-blog-publish-url` is not a directory. Does it still make sense to use `concat-to-dir`?

**Review by bastibe — CHANGES_REQUESTED** [review]
Looks like a useful addition. Could you explain what problem you encountered that this pull request solves?

**Comment by adithyaov** [comment]
> Looks like a useful addition. Could you explain what problem you encountered that this pull request solves?

Thank you for replying. 

I recently started using this package and I found that the generation of page
names was incorrect as I missed a `/` at the end of a few variables. This more
like a safety net for people like me.

In general, I think when we try to work paths we should have additional machinery
to interpret them like paths.

File: org-static-blog.el (hunk 1 above)
Comment (adithyaov) [reply org-static-blog.el:445]: Yes, `org-static-blog-publish-url` is not a directory but it has the properties
of a directory. No matter how a URL is written it should be interpreted with a
forwardslash at the end

**Review by bastibe — APPROVED** [review]

**Comment by bastibe** [comment]
Thank you very much for your contribution, then!

This does indeed remove a gotcha from using org-static-blog.

**Comment by bastibe** [comment]
Apparently this PR broke tags generation. @adithyaov, would you like to look into that and fix it?

**Comment by adithyaov** [comment]
Ah, that's unfortunate. Will send a fix along with a naive test suite.

**Pull request merged**